Features/Intel/UserInterface/VirtualKeyboardFeaturePkg/VirtualKeyboardDxe/Keyboard.c

```diff
@@ -897,9 +897,44 @@ VkApiStart (
     goto Error;
   }
 
+  for (KeyData.Key.ScanCode = SCAN_UP; KeyData.Key.ScanCode <= SCAN_F12; KeyData.Key.ScanCode++) {
+    Status = SimpleEx->RegisterKeyNotify (
+                         SimpleEx,
+                         &KeyData,
+                         VkNotifyKeyCallback,
+                         &NotifyHandle
+                         );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_VK_ROUTINE_ENTRY_EXIT | DEBUG_ERROR,
+        "ERROR - Failed to register '%c', Status: %r\n",
+        KeyData.Key.ScanCode,
+        Status
+        ));
+      break;
+    }
+  }
+
   KeyData.KeyState.KeyToggleState = 0;
   KeyData.KeyState.KeyShiftState  = 0;
   KeyData.Key.ScanCode            = SCAN_NULL;
+  for (KeyData.Key.UnicodeChar = CHAR_BACKSPACE; KeyData.Key.UnicodeChar <= CHAR_LINEFEED; KeyData.Key.UnicodeChar++) {

```

Comment:
This loop can be extended to include `CHAR_CARRIAGE_RETURN` and remove the call the `RegisterKeyNotify()` below for only `CHAR_CARRIAGE_RETURN`.

Reply:
please double check I updated the bounds correctly, I could not find the scan code definitions